File: arrays/range-query/303. Range Sum Query - Immutable/Solution.cs
public class NumArray {

    private readonly int[] _prefixSum;

    public NumArray(int[] nums) {
        _prefixSum = new int[nums.Length];
        _prefixSum[0] = nums[0];
        for (int i = 1; i < nums.Length; i++) {
            _prefixSum[i] = _prefixSum[i-1] + nums[i];
        }
    }
    
    public int SumRange(int left, int right) {
        return (left > 0) ? _prefixSum[right] - _prefixSum[left-1] : _prefixSum[right];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.SumRange(left,right);
 */

File: arrays/range-query/303. Range Sum Query - Immutable/solution.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefix_sum = [0]
        for num in nums:
            self.prefix_sum.append(num + self.prefix_sum[-1])
        

    def sumRange(self, left: int, right: int) -> int:
        return self.prefix_sum[right + 1] - self.prefix_sum[left]
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)
